site-permissions/site-permissions-impl/src/main/java/com/duckduckgo/site/permissions/impl/SitePermissionsDialogActivityLauncher.kt

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.site.permissions.impl
+
+import android.Manifest
+import android.app.AlertDialog
+import android.content.Context
+import android.net.Uri
+import android.view.LayoutInflater
+import android.view.View
+import android.webkit.PermissionRequest
+import android.widget.TextView
+import android.widget.Toast
+import androidx.activity.result.ActivityResultCaller
+import androidx.annotation.StringRes
+import com.duckduckgo.di.scopes.ActivityScope
+import com.duckduckgo.site.permissions.api.SitePermissionsDialogLauncher
+import com.duckduckgo.site.permissions.impl.R.layout
+import com.squareup.anvil.annotations.ContributesBinding
+import dagger.SingleInstanceIn
+import javax.inject.Inject
+
+@ContributesBinding(ActivityScope::class)
+@SingleInstanceIn(ActivityScope::class)
+class SitePermissionsDialogActivityLauncher @Inject constructor(
+    private val systemPermissionsHelper: SystemPermissionsHelper
+) : SitePermissionsDialogLauncher {
+
+    private lateinit var sitePermissionRequest: PermissionRequest
+    private lateinit var context: Context
+    private lateinit var permissionRequested: SitePermissionsRequestedType
+
+    override fun registerPermissionLauncher(caller: ActivityResultCaller) {
+        systemPermissionsHelper.registerPermissionLaunchers(
+            caller,
+            this::onResultSystemPermissionRequest,
+            this::onResultMultipleSystemPermissionsRequest
+        )
+    }
+
+    override fun askForSitePermission(
+        context: Context,
```

Comment:
We already have a context available that you can inject rather than passing it here